.github/workflows/tests-clustermesh-upgrade.yaml

```diff
@@ -449,6 +457,105 @@ jobs:
           cilium --context ${{ env.contextName1 }} clustermesh status --wait --wait-duration=5m
           cilium --context ${{ env.contextName2 }} clustermesh status --wait --wait-duration=5m
 
+      - name: Restart clustermesh-apiserver and ensure client can connect to new Service
+        run: |
+          echo "Restarting clustermesh-apiserver deployments"
+          kubectl --context ${{ env.contextName2 }} -n kube-system rollout restart deployment -l k8s-app=clustermesh-apiserver
+          kubectl --context ${{ env.contextName2 }} -n kube-system rollout status deployment -l k8s-app=clustermesh-apiserver
+          echo "Deploying a global Service to test failover"
+          cat << EOF > echo-failover.yaml
+          apiVersion: v1
+          kind: Service
+          metadata:
+            annotations:
+              service.cilium.io/global: "true"
+            labels:
+              kind: echo
+              context: failover
+            name: echo-other-node-failover
+            namespace: cilium-test
+          spec:
+            ipFamilies:
+            - IPv4
+            - IPv6
+            ipFamilyPolicy: PreferDualStack
+            ports:
+            - name: http
+              port: 80
+              protocol: TCP
+              targetPort: 8080
+            selector:
+              name: echo-other-node
+            sessionAffinity: None
+            type: ClusterIP
+          EOF
```

Comment:
I'd extract this into a preliminary, separate step, so that it is executed only once and we ensure consistency across the two usages.